scripts/deploy.sh

```diff
@@ -1,31 +1,58 @@
 #!/usr/bin/env bash
 
 # Exit in case of error
-set -e
+set -euo pipefail
 
-[ "${DOMAIN}" = '' ] && echo "ERROR: Remember to set DOMAIN. e.g. DOMAIN=cops-staging.openstax.org" && exit 1
-[ "${STACK_NAME}" = '' ] && echo "ERROR: Remember to set STACK_NAME. e.g. STACK_NAME=cops-stag" && exit 1
-[ "${TAG}" = '' ] && echo "ERROR: Remember to set TAG." && exit 1
+if test -n "${DEBUG_RUN_LOCAL:-}"; then
+    if test -n "${SSH_HOST:-}"; then
+        echo "ERROR: SSH_HOST and DEBUG_RUN_LOCAL should not both be set"
+        exit 111
+    fi
+    run_script() {
+        env -i HOME="$HOME" PATH="$PATH" bash -seuxo pipefail
+    }
+else
+    run_script() {
+        ssh "${SSH_HOST:?}" 'bash -seuo pipefail'
+    }
+fi
 
+: "${DOMAIN:?'Remember to set DOMAIN. e.g. DOMAIN=corgi-staging.openstax.org'}"
```

Comment:
Cool! Didn't realize this use case of the `:` operator (allowing side-effects in variable-expansion to occur)